### PR TITLE
successResultOf, failureResultOf, assertNoResult should accept coroutines

### PR DESCRIPTION
**Pull request by wsanchez**
https://twistedmatrix.com/trac/ticket/9006

### CODE REVIEW

**Comment by wsanchez** [comment]
I don't know what to do about `ubuntu16.04-py2.7-newstyle-coverage` failing…

**Comment by wsanchez** [comment]
OK fixed that, but now the pyflakes that runs in Py2.7 fails.

File: src/twisted/newsfragments/9006.feature [hunk 1]
```diff
@@ -0,0 +1,2 @@
+twisted.trial.successResultOf, twisted.trial.failureResultOf, and
```
Comment (hawkowl) [inline src/twisted/newsfragments/9006.feature:1]: these aren't actually the fqpns for these functions?

Comment (wsanchez) [reply src/twisted/newsfragments/9006.feature:1]: The fqpns are in private modules; these are (I think) the public names.

Comment (wsanchez) [reply src/twisted/newsfragments/9006.feature:1]: Like I don't think I want to say `twisted.trial._synctest._Assertions.successResultOf` here.

**Comment by glyph** [comment]
Those are… not the public names of those methods.

```
>>> from twisted.trial import successResultOf
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    from twisted.trial import successResultOf
ImportError: cannot import name 'successResultOf'
```

I think you meant `twisted.trial.unittest.SynchronousTestCase.successResultOf` ?

**Pull request merged**